Sources/BitcoinKit/Core/CoinType.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  CoinType.swift
+//  BitcoinKit
+//
+//  Copyright © 2019 BitcoinKit developers. All rights reserved.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files (the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions:
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+//
+
+import Foundation
+
+/// Registered coin types for usage in level 2 of BIP44 described in chapter "Coin type".
+public enum CoinType {
```

Comment:
It would be great to **not** use an `enum` for `CoinType`. But rather let it be a protocol, and then use a struct for coin types declared by this library. But internally also be using the protocol of course. Exactly like you did with `CoinTypeEntity `, and the static variables. Then other DLTs can use BitcoinKit. Actually we at [Radix DLT](https://www.radixdlt.com/) use it in our [Swift Library](https://github.com/radixdlt/radixdlt-swift).

So something like this:

```swift
public protocol CoinTypeConvertible {
    /// BIP44 cointype value
    var index: UInt32 { get }
    var symbol: String { get }
    var name: String { get }
}

public struct CoinType: CoinTypeConvertible {
    public let index: UInt32
    public let symbol: String
    public let name: String
}

public extension CoinType {
    static let testnet = CoinType(1, "", "Testnet (all coins)")

    static let btc = CoinType(0, "BTC", "Bitcoin")
    static let bch = CoinType(145, "BCH", "Bitcoin Cash")
}

// Ethereum and Zilliqa etc can now use BitcoinKit (both uses curve `Secp256k1`)
public extension CoinTypeEntity {
    static let ethereum = CoinType(60, "ETH", "Ether")
    static let zilliqa = CoinType(313, "ZIL", "Zilliqa")
}
```

What do you think? :)

Reply:
It makes sense.
Thank you for your feedback!

Reply:
But why do we still need CoinTypeConvertible protocol?
I think we can just have `CoinType` struct.